### PR TITLE
doc: updates for version 0.4.0

### PR DESCRIPTION
**Pull request by ru-fu**
Update the version number, add documentation link, and include
a first draft of the release notes.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

File: doc/release-notes-0.4.0.rst [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+This project is hosted by Nordic Semiconductor to demonstrate the integration of Nordic SoC support in open source projects, like MCUBoot and the Zephyr RTOS, with libraries and source code for low-power wireless applications.
+
+nRF Connect SDK v0.4.0 supports development with nRF9160 Cellular IoT devices.
+It contains references and code for Bluetooth Low Energy devices in the nRF52 Series, though development on these devices is not currently supported with the nRF Connect SDK.
+
+Highlights
+**********
+
+* Added support for the new triple mode (LTE-M/NB-IoT/GPS) with ``mfw_nrf9160_0.7.0-29.alpha`` (see :ref:`nrf9160_ug_network_mode` for instructions for changing the network mode)
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: The user guide explains only how to switch between LTE-M and NB-IoT, not GPS - should the link be included?

File: doc/release-notes-0.4.0.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+* Updated BSD library:
+
+  * Added API supporting GPS
+  * Fixed various issues (see the :ref:`nrfxlib:bsdlib_changelog`)
+
+* Added sample showing how to use the new GPS API (tested on nRF9160 DK v0.8.5)
+* Added sample showing how to download a new image via HTTP
+* Added multi-image support (see :ref:`ug_pm`)
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Is this the correct user guide?

File: doc/release-notes-0.4.0.rst [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+* Added the following samples:
+
+  * :ref:`http_application_update_sample` - demonstrates a basic FOTA (firmware over-the-air) update
+  * :ref:`mqtt_simple_sample` - connects to an MQTT broker and sends and receives data
+  * :ref:`nrf_coap_client_sample` - receives data from a public CoAP server
+  * Simple GPS sample - gets the current position and logs it on UART
+
+* Renamed the :ref:`secure_partition_manager` sample from Secure Boot to Secure Partition Manager
+* Various fixes and updates to the :ref:`nrfxlib:bsdlib` (in nrfxlib):
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Is it correct that all those fixes are in the bsdlib? @evenl

File: doc/release-notes-0.4.0.rst [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+  * Updated the library to version 0.3.0 (see the :ref:`nrfxlib:bsdlib_changelog` for details)
+  * Added support for GNSS supporting GPS as a socket
+  * Implemented :cpp:func:`bsd_os_timedwait`, allowing a proper poll operation and blocking sockets
+  * Minor fixes to the nRF91 sockets SHIM layer
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Do we have docs for the nRF91 sockets SHIM layer? (One of the subpages [here](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/bsdlib/doc/api.html)?)

File: doc/release-notes-0.4.0.rst [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+  * :ref:`lib_spm` (replacing Secure Boot) - configures security attributions for the flash, SRAM, and peripherals
+
+
+Crypto
+======
+
+* Added :ref:`nrfxlib:nrf_oberon_readme` v3.0.0 (see :ref:`nrfxlib:crypto_changelog_oberon` for details)
+* Added :ref:`nrfxlib:nrf_cc310_mbedcrypto_readme` v0.7.0 (initial, experimental release) to do hardware-accelerated cryptography using Arm CryptoCell CC310 in select architectures
+* Added :ref:`nrfxlib:nrf_security` v0.7.0 (initial, experimental release) to bridge software-implemented mbed TLS and hardware-accelerated alternatives through the nrf_cc310_mbedcrypto library
+* Added a glue layer to the :ref:`nrfxlib:nrf_security` to enable both mbed TLS APIs (software implementation) and hardware-accelerated cryptography at the same time
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Should 116 and 117 be two different items? @frkv

File: doc/release-notes-0.4.0.rst [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+Subsystems
+==========
+
+Bluetooth Low Energy
+--------------------
+
+* :ref:`peripheral_lbs` does not report the Button 1 state correctly.
+* The central samples (:ref:`central_uart`, :ref:`bluetooth_central_hids`) do not support any pairing methods with MITM protection.
+* The :ref:`gatt_dm_readme` is not supported on nRF51 devices.
+* Bluetooth LE samples cannot be built with the :ref:`nrfxlib:ble_controller` v0.1.0.
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Correct?

File: doc/release-notes-0.4.0.rst [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+* Updated to use logger instead of printk
+* Added a sample (:ref:`event_manager_sample`)
+
+Profiler
+--------
+
+* Added dynamic selection of profiled events over shell
+* Added a sample (:ref:`profiler_sample`)
+
+Build and configuration system
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Is this the right place? Or should it be under subsystems?

File: doc/release-notes-0.4.0.rst [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+DFU
+---
+
+* Firmware upgrade using Mcumgr or USB DFU is broken.
+  It is not possible to upload the image.
+
+Shell
+-----
+
+* Added a BT NUS backend.
```
Comment (ru-fu) [inline doc/release-notes-0.4.0.rst]: Is this change merged? @grochu @nordic-krch @jarz-nordic

Comment (jarz-nordic) [reply doc/release-notes-0.4.0.rst]: @ru-fu : No ! I apologize and thank you for finding.

File: doc/release-notes-0.4.0.rst (hunk 6 above)
Comment (shantha-14) [reply doc/release-notes-0.4.0.rst]: yes

File: doc/release-notes-0.4.0.rst (hunk 8 above)
Comment (ru-fu) [reply doc/release-notes-0.4.0.rst]: removed

File: doc/release-notes-0.4.0.rst (hunk 6 above)
Comment (grochu) [reply doc/release-notes-0.4.0.rst]: I think it's correct. _BTW one could ask why do we provide this BLE Controller at all in 0.4.0_?

File: doc/release-notes-0.4.0.rst (hunk 1 above)
Comment (MonikaLieLarsen) [reply doc/release-notes-0.4.0.rst]: Don't think that is strictly necessary, since next bullet point says there is a sample on how to use GPS.

**Comment by shantha-14** [comment]
Segger Jlink version is 6.41

File: doc/release-notes-0.4.0.rst (hunk 2 above)
Comment (MonikaLieLarsen) [reply doc/release-notes-0.4.0.rst]: looks like it? @hakonfam

**Comment by Raane** [comment]
Additional point @lemrey noticed:

•	Firmware upgrade using Mcumgr or USB DFU is broken. It is not possible to upload the image.

Consider if this is something we can include last minute.

File: doc/release-notes-0.4.0.rst (hunk 4 above)
Comment (rlubos) [reply doc/release-notes-0.4.0.rst]: No, not really, it's an integration layer between `bsdlib` and Zephyr, namely [this](https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/lib/bsdlib/nrf91_sockets.c) file. A very brief documentation of the API it's implementing is present in Zephyr: https://docs.zephyrproject.org/latest/reference/networking/net_offload.html#socket-offloading

**Comment by ru-fu** [comment]
> 
> 
> Additional point @lemrey noticed:
> 
> • Firmware upgrade using Mcumgr or USB DFU is broken. It is not possible to upload the image.
> 
> Consider if this is something we can include last minute.

See line 244 - is this the wrong place for it?

**Comment by ru-fu** [comment]
From the changelog page in the repo:

> nRF91: The previous nrf9160_pca10090 is split into two boards, one for the secure (nrf9160_pca10090s) and one for the non-secure (nrf9160_pca10090ns) image. They both reside in the same board folder.
> nRF91: Deprecated the nrf91 branch in https://github.com/NordicPlayground/fw-nrfconnect-zephyr in favor of master, which now supports the nRF9160 IC

This has been changed for a while, but should it be included?

File: doc/release-notes-0.4.0.rst (hunk 1 above)
Comment (evenl) [reply doc/release-notes-0.4.0.rst]: The AT command document also explains how the "AT%XSYSTEMMODE" and "AT%XMAGPIO" commands works. Maybe refer to this documentation, or do the nrf9160_ug_network_mode section already mention this document?

Comment (ru-fu) [reply doc/release-notes-0.4.0.rst]: Yes, the nrf9160_ug_network_mode section points to the system mode section in the AT commands documentation. So I guess this is the best we can do for now. :)

File: doc/release-notes-0.4.0.rst (hunk 3 above)
Comment (ru-fu) [reply doc/release-notes-0.4.0.rst]: Section updated as follows:
image

File: doc/release-notes-0.4.0.rst (hunk 4 above)
Comment (ru-fu) [reply doc/release-notes-0.4.0.rst]: OK, wording updated and clarified where the file is located.

**Comment by shantha-14** [comment]
Known Issue: BLE examples: throughput disconnects when executed between two  nRF51 boards.

**Comment by lemrey** [comment]
I see it's still missing:
@ru-fu please mention under nrf9160:
- Fixed an issue in the LTE BLE Gateway sample where the host and the controller would go out of sync after a reset.

> > Additional point @lemrey noticed:
> > • Firmware upgrade using Mcumgr or USB DFU is broken. It is not possible to upload the image.
> > Consider if this is something we can include last minute.
> 
> See line 244 - is this the wrong place for it?

No, that's not what I meant, clearly that's already mentioned.
I meant to inquiry as to which boards are affected, perhaps add a bit of detail as to what is failing.
Who reported this? Is it an issue with MCUboot or.. ? @hakonfam .

File: doc/release-notes-0.4.0.rst (hunk 7 above)
Comment (ru-fu) [reply doc/release-notes-0.4.0.rst]: Seems to be ok.

**Comment by grochu** [comment]
> Known Issue: BLE examples: throughput disconnects when executed between two nRF51 boards.

This is already covered by: `* The :ref:`gatt_dm_readme` is not supported on nRF51 devices.`
I phrased it like this, because an sample which uses the GATT Discovery Manager on nRF51 (Cortex M0) will crash - which is also why  Throughput disconnects (timeout due to other board crashing).

File: doc/release-notes-0.4.0.rst (hunk 5 above)
Comment (tejlmand) [reply doc/release-notes-0.4.0.rst]: Rephrased, combining the two items:
```suggestion
* Added :ref:`nrfxlib:nrf_security` v0.7.0 (initial, experimental release) to bridge software-implemented mbed TLS and hardware-accelerated alternatives, nrf_cc310_mbedcrypto library, through the use of a glue layer
```

**Pull request merged**